frontend/src/api/k8s/instructlab.ts

```diff
@@ -0,0 +1,19 @@
+import { k8sPatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { DataScienceClusterKind } from '~/k8sTypes';
+import { DataScienceClusterModel } from '~/api/models';
+
+export const toggleInstructLabState = (
+  instructLabState: string,
+  dscName: string,
+): Promise<DataScienceClusterKind> =>
+  k8sPatchResource<DataScienceClusterKind>({
+    model: DataScienceClusterModel,
+    queryOptions: { name: dscName },
+    patches: [
+      {
+        op: 'replace',
+        path: '/spec/components/datasciencepipelines/managedPipelines/instructLab/state',
+        value: instructLabState,
+      },
+    ],
+  });
```

Comment:
this whole file should be inside the k8s/dsc.ts file. there is no specific ilab k8s api